test/LoggerTest.php

```diff
@@ -0,0 +1,147 @@
+        $this->mockhandler2->addFilter(new MessageFilter('/notwokringregexpattern/'));
+        $this->handlers[] = $this->mockhandler1;
+        $this->handlers[] = $this->mockhandler2;
+
+
+        $this->logging = new Logger($this->handlers, null, $this->messagefilter);
+    }
+
+    public function testSerializeAndAddFilter(): void
+    {
+        $data = $this->logging->serialize();
+        $this->assertStringContainsString('MockHandler', $data);
+        $this->assertStringContainsString('MessageFilter', $data);
+        $this->assertStringContainsString('/emergency/', $data);
+    }
+
+    public function testSerializeAndAddHandler(): void
+    {
+        $mockhandler3 = new MockHandler();
+        $mockhandler3->addFilter(new MessageFilter('/whazzaaa/'));
+        $this->logging->addHandler($mockhandler3);
+        $data = $this->logging->serialize();
+        $this->assertStringContainsString('/whazzaaa/', $data);
+    }
+
+    public function testUnserialize(): void
+    {
+        $data = $this->logging->serialize();
+        $this->assertNull($this->logging->unserialize($data));
+    }
+
+    public function testErrorsUnserialize(): void
+    {
+        $data[] = ['this is not a string'];
+        $data[] = serialize('blabla');
+        $data[] = serialize([]);
+        $data[] = serialize(['another version']);
+
+        foreach ($data as $value) {
+            try {
+                $this->logging->unserialize($value);
+            } catch (\Throwable $th) {
+                $this->assertInstanceOf(LogException::class, $th);
+            }
+        }
+    }
+
+    public function testLogMethodsByCheckingTheirClassAndIfMockHandlerWorks()
+    {
+        $methods = get_class_methods(Logger::class);
+
+        $loggerinterface_methods = array_flip(get_class_methods(LoggerInterface::class));
+
+        foreach ($methods as $key => $method) {
+            if (array_key_exists($method, $loggerinterface_methods)) {
+                if ($method == 'log') {
+                    $this->logging->$method($this->level1, $this->message1);
+                    $this->assertEquals($this->mockhandler1->check->message(), $this->logMessage1->message());
+                    $this->assertEquals($this->mockhandler1->check->level()->name(), $this->logMessage1->level()->name());
+                } else {
+                    $this->logging->$method($this->message1);
+                    $this->assertEquals($this->mockhandler1->check->message(), $this->logMessage1->message());
+                    $this->assertEquals($this->mockhandler1->check->level()->name(), $method);
+                }
+            }
+        }
+    }
+
+    public function testStringAsMessageForLog()
+    {
+        $this->assertNull($this->logging->log($this->level1, "righty o, this should be a message for an alert warrrninggg"));
+    }
+
+    public function testFiltersOfLoggerWorkWithoutMockHandlerFilter()
+    {
+        $this->mockhandler2->log($this->logMessage1);
```

Comment:
This would be much more readable if you defined these objects here, instead of in the setup method

Reply:
Ok, I added local variables and rewrote the test